nima/tests/integration/webclient/webclient/src/test/java/io/helidon/nima/webclient/http1/ClientRequestImplTest.java

```diff
@@ -0,0 +1,401 @@
+    }
+
+    private static void validateFailedResponse(Http1Client client, String errorMessage) {
+        String requestEntity = "Sending Something";
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        IllegalStateException ie = assertThrows(IllegalStateException.class, () -> request.submit(requestEntity));
+        assertThat(ie.getMessage(), containsString(errorMessage));
+    }
+
+    private static void validateChunkTransfer(Http1ClientResponse response, boolean chunked, long contentLength, String entity) {
+        assertThat(response.status(), is(Http.Status.OK_200));
+        if (contentLength == NO_CONTENT_LENGTH) {
+            assertThat(response.headers(), noHeader(REQ_CONTENT_LENGTH_HEADER_NAME));
+        } else {
+            assertThat(response.headers(), hasHeader(REQ_CONTENT_LENGTH_HEADER_NAME, String.valueOf(contentLength)));
+        }
+        if (chunked) {
+            assertThat(response.headers(), hasHeader(REQ_CHUNKED_HEADER));
+        } else {
+            assertThat(response.headers(), noHeader(REQ_CHUNKED_HEADER.headerName()));
+        }
+        String responseEntity = response.entity().as(String.class);
+        assertThat(responseEntity, is(entity));
+    }
+
+    private static void responseHandler(ServerRequest req, ServerResponse res) throws IOException {
+        customHandler(req, res, false);
+    }
+
+    private static void chunkResponseHandler(ServerRequest req, ServerResponse res) throws IOException {
+        customHandler(req, res, true);
+    }
+
+    private static void customHandler(ServerRequest req, ServerResponse res, boolean chunkResponse) throws IOException {
+        Headers reqHeaders = req.headers();
+        if (reqHeaders.contains(Http.HeaderValues.EXPECT_100)) {
+            res.headers().set(REQ_EXPECT_100_HEADER_NAME);
+        }
+        if (reqHeaders.contains(Http.Header.CONTENT_LENGTH)) {
+            res.headers().set(REQ_CONTENT_LENGTH_HEADER_NAME, reqHeaders.get(Http.Header.CONTENT_LENGTH).value());
+        }
+        if (reqHeaders.contains(Http.HeaderValues.TRANSFER_ENCODING_CHUNKED)) {
+            res.headers().set(REQ_CHUNKED_HEADER);
+        }
+
+        try (InputStream inputStream = req.content().inputStream();
+                OutputStream outputStream = res.outputStream()) {
+            if (!chunkResponse) {
+                new ByteArrayInputStream(inputStream.readAllBytes()).transferTo(outputStream);
+            } else {
+                // Break the entity into 3 parts and send them in chunks
+                int chunkParts = 3;
+                byte[] entity = inputStream.readAllBytes();
+                int regularChunkLen = entity.length / chunkParts;
+                int lastChunkLen = regularChunkLen + entity.length % chunkParts;
+                for (int i = 0; i < chunkParts; i++) {
+                    int chunkLen = i != (chunkParts - 1) ? regularChunkLen : lastChunkLen;
```

Comment:
Minor comment, may be more readable "int chunkLen = (i != chunkParts - 1) ? ..."

Reply:
Done

Reply:
Oops, first change was not as suggested. Removed unneeded parentheses to conform with the suggestion.